ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -1779,5 +1842,53 @@ mod tests {
             ";
             test_with_spillalloc(&m, &patt_lines);
         }
+
+        #[test]
+        fn cg_srem() {
+            let mut m = test_module();
+            let i8_ty_idx = m.insert_ty(jit_ir::Ty::Integer(IntegerTy::new(8))).unwrap();
+            let op1 = m
+                .push_and_make_operand(jit_ir::LoadTraceInputInst::new(0, i8_ty_idx).into())
+                .unwrap();
+            let op2 = m
+                .push_and_make_operand(jit_ir::LoadTraceInputInst::new(1, i8_ty_idx).into())
+                .unwrap();
+            m.push(jit_ir::SRemInst::new(op1, op2).into()).unwrap();
+            let patt_lines = "
+                ...
+                ; %2: i8 = srem %0, %1
+                ... xor rdx, rdx
+                ... movzx rax, byte ptr [rbp-0x01]
+                ... movzx r13, byte ptr [rbp-0x02]
+                ... idiv r13b
+                ... mov [rbp-0x03], dl
+                ...
+            ";
+            test_with_spillalloc(&m, &patt_lines);
+        }
+
+        #[test]
+        fn cg_trunc() {
+            let mut m = test_module();
+            let i8_ty_idx = m.insert_ty(jit_ir::Ty::Integer(IntegerTy::new(8))).unwrap();
+            let i32_ty_idx = m
+                .insert_ty(jit_ir::Ty::Integer(IntegerTy::new(32)))
+                .unwrap();
+            let op = m
+                .push_and_make_operand(jit_ir::LoadTraceInputInst::new(0, i32_ty_idx).into())
+                .unwrap();
+            m.push(jit_ir::TruncInst::new(&op, i8_ty_idx).into())
+                .unwrap();
+            let patt_lines = "
+                ...
+                ; %0: i32 = load_ti 0, i32
```

Comment:
I'm about to raise a PR which removes the trailing `, T` from `load_ti` (see https://github.com/ykjit/yk/commit/24bcb9fec9eb5d3180f2fbe15438d197ff007b0c). It might be easiest to remove those from this PR, as *probably* because of the ykllvm pr, the load_ti PR will end up going in first. You'll only want to edit the `load_ti`s introduced in this PR, though, which I think is just this one and the one two lines below.

Reply:
Ok, I'll wait for you stuff to go in first then.

Reply:
If you want you can do this preemptively now with a commit. By the looks of things it's the only comment we have anyway!